pkg/logevents/logevents.go

```diff
@@ -0,0 +1,87 @@
+/*
+* Copyright 2020 Armory, Inc.
+
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+
+*    http://www.apache.org/licenses/LICENSE-2.0
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package logevents
+
+import (
+	"encoding/json"
+	"github.com/armory/dinghy/pkg/cache"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+type LogEventsClient interface {
+	GetLogEvents() ([]LogEvent, error)
+	SaveLogEvent(logEvent LogEvent) error
+}
+
+type LogEventRedisClient struct {
+	RedisClient *cache.RedisCache
+}
+
+type LogEvent struct {
+	Org 		string
+	Repo 		string
+	Files		[]string
+	Message		string
+	Date		int64
+	Commits		[]string
+}
+
+
+
+func (c LogEventRedisClient) GetLogEvents() ([]LogEvent, error) {
+	loge := log.WithFields(log.Fields{"func": "GetLogEvents"})
+	keys, _, err := c.RedisClient.Client.Scan(0, cache.CompileKey("logEvent*"), 1000).Result()
```

Comment:
Is it correct to say that we're limiting ourselves to 1000 records? Does the Result struct indicate if there are more records to pull?

Reply:
This was fixed. Now all records are retrieved.